packages/flutter_tools/lib/src/asset.dart

```diff
@@ -53,7 +57,8 @@ abstract class AssetBundleFactory {
     @required Logger logger,
     @required FileSystem fileSystem,
     @required Platform platform,
-  }) => _ManifestAssetBundleFactory(logger: logger, fileSystem: fileSystem, platform: platform);
```

Comment:
Since we can't build a split aot in debug mode, could we structure this so that the calling code always passes false for debug and otherwise leaves the build mode out of the asset system?